tests/conftest.py

```diff
@@ -1,80 +1,251 @@
-# directory-specific hook implementations
 import os
-import shutil
 import sys
-import tempfile
+import warnings
+from contextlib import nullcontext
+from pathlib import Path
 
 import datajoint as dj
+import pynwb
+import pytest
+from datajoint.logging import logger
 
-from .datajoint._config import DATAJOINT_SERVER_PORT
-from .datajoint._datajoint_server import (
-    kill_datajoint_server,
-    run_datajoint_server,
-)
+from .container import DockerMySQLManager
 
-thisdir = os.path.dirname(os.path.realpath(__file__))
-sys.path.append(thisdir)
+# ---------------------- CONSTANTS ---------------------
 
-
-global __PROCESS
-__PROCESS = None
+# globals in pytest_configure: BASE_DIR, SERVER, TEARDOWN, VERBOSE
+warnings.filterwarnings("ignore", category=UserWarning, module="hdmf")
 
 
 def pytest_addoption(parser):
+    """Permit constants when calling pytest at command line
+
+    Example
+    -------
+    > pytest --quiet-spy
+
+    Parameters
+    ----------
+    --quiet-spy (bool):  Default False. Allow print statements from Spyglass.
+    --no-teardown (bool): Default False. Delete pipeline on close.
+    --no-server (bool): Default False. Run datajoint server in Docker.
+    --datadir (str): Default './tests/test_data/'. Dir for local input file.
+        WARNING: not yet implemented.
+    """
+    parser.addoption(
+        "--quiet-spy",
+        action="store_true",
+        dest="quiet_spy",
+        default=False,
+        help="Quiet logging from Spyglass.",
+    )
+    parser.addoption(
+        "--no-server",
+        action="store_true",
+        dest="no_server",
+        default=False,
+        help="Do not launch datajoint server in Docker.",
+    )
     parser.addoption(
-        "--current",
+        "--no-teardown",
         action="store_true",
-        dest="current",
         default=False,
-        help="run only tests marked as current",
+        dest="no_teardown",
+        help="Tear down tables after tests.",
+    )
+    parser.addoption(
+        "--base-dir",
+        action="store",
+        default="./tests/_data/",
+        dest="base_dir",
+        help="Directory for local input file.",
     )
 
 
 def pytest_configure(config):
-    config.addinivalue_line(
-        "markers", "current: for convenience -- mark one test as current"
+    global BASE_DIR, SERVER, TEARDOWN, VERBOSE
+
+    TEARDOWN = not config.option.no_teardown
+    VERBOSE = not config.option.quiet_spy
+
+    BASE_DIR = Path(config.option.base_dir).absolute()
+    BASE_DIR.mkdir(parents=True, exist_ok=True)
+    os.environ["SPYGLASS_BASE_DIR"] = str(BASE_DIR)
+
+    SERVER = DockerMySQLManager(
```

Comment:
Starts docker server, global object in case we need to check status. circumvents use of cachery script execution tools to monitor docker via bash